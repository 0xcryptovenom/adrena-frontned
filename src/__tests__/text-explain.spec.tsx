import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';

import TextExplain from '@/components/common/TextExplain/TextExplain';

describe('text-explain.tsx', () => {
  it('renders', () => {
    const { container, getByText } = render(<TextExplain title="Testing" />);
    expect(container).toMatchSnapshot();
  });

  it('renders the provided title prop to the DOM', () => {
    const { getByText } = render(<TextExplain title="Testing" />);
    expect(getByText('Testing')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<TextExplain title="Testing" />);
    expect(container).toMatchSnapshot();
  });
});
