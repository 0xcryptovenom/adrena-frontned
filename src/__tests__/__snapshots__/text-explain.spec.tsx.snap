// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`text-explain.tsx > matches snapshot 1`] = `
<div>
  <div
    class="absolute left-0 w-full z-20 flex flex-col justify-center items-center top-[-1em]"
  >
    <span
      class="text-gray-500 whitespace-nowrap"
    >
      Testing
    </span>
    <div
      class="h-1 border-l-2 border-r-2 border-gray-600 w-full border-t-2 mt-[0.15em]"
    />
  </div>
</div>
`;

exports[`text-explain.tsx > renders 1`] = `
<div>
  <div
    class="absolute left-0 w-full z-20 flex flex-col justify-center items-center top-[-1em]"
  >
    <span
      class="text-gray-500 whitespace-nowrap"
    >
      Testing
    </span>
    <div
      class="h-1 border-l-2 border-r-2 border-gray-600 w-full border-t-2 mt-[0.15em]"
    />
  </div>
</div>
`;
